invesalius/presets.py

```diff
@@ -15,108 +15,115 @@
 #    COMERCIALIZACAO ou de ADEQUACAO A QUALQUER PROPOSITO EM
 #    PARTICULAR. Consulte a Licenca Publica Geral GNU para obter mais
 #    detalhes.
-#--------------------------------------------------------------------------
+# --------------------------------------------------------------------------
 import glob
 import os
 import plistlib
-
-import invesalius.constants as const
-
-from invesalius.pubsub import pub as Publisher
+import typing
 
 from invesalius import inv_paths
+from invesalius.i18n import tr
+from invesalius.pubsub import pub as Publisher
 from invesalius.utils import TwoWaysDictionary
-from invesalius.i18n import tr as _
 
-class Presets():
+if typing.TYPE_CHECKING:
+    from pathlib import Path
 
+
+class Presets:
     def __init__(self):
-        self.thresh_ct = TwoWaysDictionary({
-            _("Bone"):(226,3071),
-            _("Soft Tissue"):(-700,225),
-            _("Enamel (Adult)"):(1553,2850),
-            _("Enamel (Child)"):(2042,3071),
-            _("Compact Bone (Adult)"):(662,1988),
-            _("Compact Bone (Child)"):(586,2198),
-            _("Spongial Bone (Adult)"):(148,661),
-            _("Spongial Bone (Child)"):(156,585),
-            _("Muscle Tissue (Adult)"):(-5,135),
-            _("Muscle Tissue (Child)"):(-25,139),
-            _("Fat Tissue (Adult)"):(-205,-51),
-            _("Fat Tissue (Child)"):(-212,-72),
-            _("Skin Tissue (Adult)"):(-718,-177),
-            _("Skin Tissue (Child)"):(-766,-202), 
-            _("Custom"):(0, 0)
-        })
-
-        self.thresh_mri = TwoWaysDictionary({
-            _("Bone"):(1250,4095),
-            _("Soft Tissue"):(324,1249),
-            _("Enamel (Adult)"):(2577,3874),
-            _("Enamel (Child)"):(3066,4095),
-            _("Compact Bone (Adult)"):(1686,3012),
-            _("Compact Bone (Child)"):(1610,3222),
-            _("Spongial Bone (Adult)"):(1172,1685),
-            _("Spongial Bone (Child)"):(1180,1609),
-            _("Muscle Tissue (Adult)"):(1019,1159),
-            _("Muscle Tissue (Child)"):(999,1163),
-            _("Fat Tissue (Adult)"):(819,973),
-            _("Fat Tissue (Child)"):(812,952),
-            _("Skin Tissue (Adult)"):(306,847),
-            _("Skin Tissue (Child)"):(258,822), 
-            _("Custom"):(0, 0)
-        })
+        self.thresh_ct = TwoWaysDictionary(
+            {
+                tr("Bone"): (226, 3071),
```

Comment:
ruff replaced _ with tr?

Reply:
No, This is actually a change that I manually added. There are many reasons for not using the special underscore character '_' for normal variable names in python. All of the following reasons can be found in [Real Python](https://realpython.com/python-double-underscore/#other-usages-of-underscores-in-python)

1. '\_' by convention means unused local variable name and will be automatically ignored by linters and type checkers if it is not used. Even though in this context the imported variable is global, assigning a variable to '\_' and then using it violates this convention and can create confusion.
![image](https://github.com/invesalius/invesalius3/assets/113486195/bcb25a69-951f-4819-8592-8edd6d38dfbb)
I personally can use '\_' as a throwaway variable in loops and it can be confusing sometimes if I tried to use '\_' again in the same scope. 
![image](https://github.com/invesalius/invesalius3/assets/113486195/96c764f4-6c77-4eab-bcbc-7b08fccdfbd9)
1. In Interactive python shell sessions, '\_' is a special variable that stores the last calculated result. This may seem irrelevant but you can actually debug python scripts using `python -i foo.py`.
![image](https://github.com/invesalius/invesalius3/assets/113486195/9e151273-765f-42bd-a1dc-3cdc6e53325f)
Therefore, You shouldn't look for a variable called '\_' in an interactive session.
1. Structural pattern matching was introduced to Python in version 3.10. It uses a match … case construct to compare an object to several different cases. Such statements are effective at deconstructing data structures and picking out individual elements. Python treats a single underscore as a wildcard in a case statement, so it’ll match anything. You often use it to alert the user that an object doesn’t have the expected structure.
```console
>>> def sum_list(numbers):
...     match numbers:
...         case []:
...             return 0
...         case [int(first) | float(first), *rest]:
...             return first + sum_list(rest)
...         case _:
...             raise ValueError(f"can only sum lists of numbers")
...
>>> sum_list([1, 2, 3])
6
>>> sum_list(["x", "y"])
Traceback (most recent call last):
    ...
ValueError: can only sum lists of numbers
```
The above behaviour kind of makes '\_' as a soft keyword in python. This is more of an incentive to not use '\_' for variable names without a really good reason to do.

I recommend that all instances of `from invesalius.i18n import tr as \_` are changed to `from invesalius.i18n import tr`. Please tell me what do you think about this issue?

Reply:
I prefer `tr` too. We used `_` because of [gettext doc](https://docs.python.org/3/library/gettext.html). But it's possible to use other name. In Qt the use `tr` too. But it'll be necessary to change `_` to `tr` in the whole InVesalius source code. I know VSCode has a symbol rename function that can help here. Neovim and Helix have too.

Reply:
Done.
I used the vscode shortcut F2 to change _ var name to tr in all files. Then I used find and replace all function to change all instances of `from invesalius.i18n import tr as tr` to `from invesalius.i18n import tr `.